homeassistant/components/huawei_lte/device_tracker.py

```diff
@@ -1,63 +1,52 @@
 """Support for device tracking of Huawei LTE routers."""
 
 import logging
-from typing import Any, Dict, List, Optional
 
 import attr
-import voluptuous as vol
 
-import homeassistant.helpers.config_validation as cv
-from homeassistant.components.device_tracker import PLATFORM_SCHEMA, DeviceScanner
+from homeassistant.components.device_tracker import SOURCE_TYPE_ROUTER
 from homeassistant.const import CONF_URL
-from . import RouterData
+from homeassistant.helpers.event import track_time_interval
+from . import MIN_TIME_BETWEEN_UPDATES, RouterData
 from .const import DOMAIN, KEY_WLAN_HOST_LIST
 
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Optional(CONF_URL): cv.url})
 
-HOSTS_PATH = f"{KEY_WLAN_HOST_LIST}.Hosts.Host"
+def setup_scanner(hass, config, see, discovery_info=None):
```

Comment:
This is the legacy interface. For config entries we need to use `async_setup_entry`.

Reply:
I remember having a look at this, but IIRC didn't figure out how would I go about making the tracker available also via good old yaml support, not just config entries. The current implementation at least works with both.

Reply:
We'll import the yaml into a config flow and save it as a config entry. This is the current standard to support both config yaml and config flow. There's no option to keep this legacy interface and start using config entries.

Reply:
The components that do this that hit my quick grep seem to do it only once. I guess this means that after the import, further changes need to be made through the UI, and YAML changes no longer have any effect. Yet none of them document this behavior in the component docs. This UX doesn't sound good to me. Have I misunderstood something?

Reply:
You're right that currently many integrations that import the yaml config into a config entry doesn't allow syncing a change to config yaml with the created config entry. We currently don't have a helper or recommendation here. But it's totally possible to do this if wanted.

Eg, checksum the imported yaml/dict and save that checksum in the config entry. On next start, calculate the checksum of the yaml config again and check if we need to update the config entry.